tests/src/utils/fs.test.ts

```diff
@@ -70,69 +31,118 @@ describe('normalizePath', () => {
+		);
+		mockFs.restore();
 	});
 
 	test('should return null with invalid json file', async () => {
-		await expect(readJsonFile('.invalid-config.json')).resolves.toEqual(null);
+		mockFs({
+			'invalid.json': 'invalid-file',
+		});
+		await expect(readJsonFile('invalid.json')).resolves.toEqual(null);
+		mockFs.restore();
 	});
 });
 
 describe('validateFile', () => {
+	beforeAll(() => {
+		mockFs({
+			'functions/index.func': { 'index.js': 'valid-file' },
+		});
+	});
+	afterAll(() => {
+		mockFs.restore();
+	});
 	test('valid file returns true', async () => {
 		await expect(
-			validateFile('validTest/functions/index.func/index.js')
+			validateFile('functions/index.func/index.js')
 		).resolves.toEqual(true);
 	});
 
 	test('valid directory returns false', async () => {
-		await expect(
-			validateFile('validTest/functions/index.func')
-		).resolves.toEqual(false);
+		await expect(validateFile('functions/index.func')).resolves.toEqual(false);
 	});
 
 	test('invalid path returns false', async () => {
-		await expect(validateFile('invalidTest/invalidPath')).resolves.toEqual(
-			false
-		);
+		await expect(
+			validateFile('functions/invalid-index.func/index.js')
+		).resolves.toEqual(false);
 	});
 });
 
 describe('validateDir', () => {
+	beforeAll(() => {
+		mockFs({
+			'functions/index.func': { 'index.js': 'valid-file' },
+		});
+	});
+	afterAll(() => {
+		mockFs.restore();
+	});
 	test('valid directory returns true', async () => {
-		await expect(
-			validateDir('validTest/functions/index.func')
-		).resolves.toEqual(true);
+		mockFs({
+			'functions/index.func': { 'index.js': 'valid-file' },
+		});
+		await expect(validateDir('functions/index.func')).resolves.toEqual(true);
+		mockFs.restore();
 	});
 
 	test('valid file returns false', async () => {
-		await expect(
-			validateDir('validTest/functions/index.func/index.js')
-		).resolves.toEqual(false);
+		await expect(validateDir('functions/index.func/index.js')).resolves.toEqual(
+			false
+		);
 	});
 
 	test('invalid path returns false', async () => {
-		await expect(validateDir('invalidTest/invalidPath')).resolves.toEqual(
-			false
-		);
+		await expect(validateDir('invalidPath')).resolves.toEqual(false);
 	});
 });
 
 describe('readPathsRecursively', () => {
+	beforeAll(() => {
+		mockFs({
+			root: {
+				functions: {
+					'(route-group)': {
+						'page.func': {
+							'index.js': 'page-js-code',
+						},
+					},
+					'index.func': {
+						'index.js': 'index-js-code',
+					},
+					'home.func': {
+						'index.js': 'home-js-code',
+					},
+				},
+			},
+		});
+	});
+	afterAll(() => {
+		mockFs.restore();
+	});
 	test('should read all paths recursively', async () => {
-		await expect(readPathsRecursively('validTest/functions')).resolves.toEqual([
-			'validTest/functions/(route-group)/page.func/index.js',
-			'validTest/functions/index.func/index.js',
-			'validTest/functions/home.func/index.js',
-		]);
+		const paths = await readPathsRecursively('root/functions');
```

Comment:
```suggestion
		const paths = (await readPathsRecursively('root/functions')).map(path =>
			normalizePath(path)
		);
```
the paths need to be normalized for the tests to pass for Windows users